src/Sulu/Bundle/AutomationBundle/Controller/TaskController.php

```diff
@@ -0,0 +1,203 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\AutomationBundle\Controller;
+
+use Doctrine\ORM\EntityManagerInterface;
+use FOS\RestBundle\Routing\ClassResourceInterface;
+use JMS\Serializer\DeserializationContext;
+use Sulu\Bundle\AutomationBundle\Entity\Task;
+use Sulu\Bundle\AutomationBundle\Tasks\Manager\TaskManagerInterface;
+use Sulu\Component\Rest\ListBuilder\FieldDescriptorInterface;
+use Sulu\Component\Rest\ListBuilder\ListRepresentation;
+use Sulu\Component\Rest\RestController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/**
+ * Provides api for tasks.
+ */
+class TaskController extends RestController implements ClassResourceInterface
+{
+    /**
+     * Returns list of tasks.
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function cgetAction(Request $request)
+    {
+        $restHelper = $this->get('sulu_core.doctrine_rest_helper');
+        $factory = $this->get('sulu_core.doctrine_list_builder_factory');
+        $fieldDescriptors = $this->getFieldDescriptors();
+
+        $listBuilder = $factory->create(Task::class);
+        $restHelper->initializeListBuilder($listBuilder, $fieldDescriptors);
+
+        if ($entityClass = $request->get('entity-class')) {
+            $listBuilder->where($fieldDescriptors['entityClass'], $entityClass);
+        }
+        if ($entityId = $request->get('entity-id')) {
+            $listBuilder->where($fieldDescriptors['entityId'], $entityId);
+        }
+
+        return $this->handleView(
+            $this->view(
+                new ListRepresentation(
+                    $listBuilder->execute(),
+                    'tasks',
+                    'get_tasks',
+                    $request->query->all(),
+                    $listBuilder->getCurrentPage(),
+                    $listBuilder->getLimit(),
+                    $listBuilder->count()
+                )
+            )
+        );
+    }
+
+    /**
+     * Returns task for given id.
+     *
+     * @param int $id
+     *
+     * @return Response
+     */
+    public function getAction($id)
+    {
+        $manager = $this->getTaskManager();
+        $task = $manager->findById($id);
+
+        if (!$task) {
+            throw new NotFoundHttpException();
```

Comment:
Actually we changed that to using the `FOSRestBundle` exception functionality for that. I would throw a `TaskNotFoundException` in the `findById` method, and configure the corresponding http error code in the FOSRestBundle configuration.